src/main/host/host.rs

```diff
@@ -877,6 +877,19 @@ impl Host {
         unsafe { &*self.shim_shmem.get() }
     }
 
+    /// Returns the specified thread if it exists. If you already have the thread's process,
+    /// [`Process::thread_borrow`] may be more efficient.
+    pub fn thread(&self, virtual_tid: ThreadId) -> Option<RootedRc<RootedRefCell<Thread>>> {
```

Comment:
My thinking is that it's nicer to just clone the `RootedRc` and add the explicit drop than to add more indirection and unsafe code to workaround the multiple levels of borrowing, at least in this case. In terms of performance there should be effectively no difference. Even if it's replaced with `Arc` in the future I don't think an extra atomic inc/dec would be bad since it's unlikely that you'd call this method very often, and we could just add a note about it to the docstring.

> Maybe this could at least be named something that makes it clearer that the caller is getting an owned `RootedRc` back, though? I don't remember if we already have a naming convention for such methods; maybe `thread_cloned_rc`?

Why should this be in the function name? Is this just to remind the caller that they need to do the explicit drop since it's a `RootedRc`? I have no problem changing it though, so I'll change it to `thread_cloned_rc`.

Reply:
> Why should this be in the function name? Is this just to remind the caller that they need to do the explicit drop since it's a RootedRc?

Yeah, that's the idea